src/test/java/openfasttrack/importer/markdown/TestImportMarkdown.java

```diff
@@ -217,4 +180,72 @@ public void testSingleNeeds()
         runImporterOnText(singleNeedsItem);
         verify(this.listenerMock, times(1)).addNeededArtifactType(NEEDS_ARTIFACT_TYPE1);
     }
+
+    // [utest->dsn~md.eb-markdown-id~1]
+    @Test
+    public void testIdentifyLegacyId()
+    {
+        assertMatch(MdPattern.ID, "a:b, v0", "req:test, v1", "req:test,v1", "req:test, v999",
+                "req:test.requirement, v1", "req:test_underscore, v1",
+                "`req:test1, v1`arbitrary text");
+        assertMismatch(MdPattern.ID, "test, v1", "req-test, v1", "req.4test, v1");
+    }
+
+    @Test
+    public void testFindLegacyRequirement()
+    {
+        final String completeItem = createCompleteSpecificationItemInLegacyMarkdownFormat();
+        runImporterOnText(completeItem);
+        assertAllImporterEventsForLegacyItemCalled();
+    }
+
+    // [utest->dsn~md.needs-coverage-list~2]
+    private String createCompleteSpecificationItemInLegacyMarkdownFormat()
+    {
+        return "# " + TITLE //
+                + "\n" //
+                + "`" + LEGACY_ID + "`" //
+                + "\nDescription:\n" + DESCRIPTION_LINE1 + "\n" //
+                + DESCRIPTION_LINE2 + "\n" //
+                + DESCRIPTION_LINE3 + "\n" //
+                + "\nRationale:\n" //
+                + RATIONALE_LINE1 + "\n" //
+                + RATIONALE_LINE2 + "\n" //
+                + "\nDepends:\n\n" //
+                + "  + " + LEGACY_DEPENDS_ON_ID1 + "\n" //
+                + "  - " + LEGACY_DEPENDS_ON_ID2 + "\n" //
+                + "\nCovers:\n\n" //
+                + "  * " + LEGACY_COVERED_ID1 + "\n" //
+                + " + " + LEGACY_COVERED_ID2 + "\n" //
+                + "\nComment:\n\n" //
+                + COMMENT_LINE1 + "\n" //
+                + COMMENT_LINE2 + "\n" //
+                + "\nNeeds:\n" //
+                + "   * " + NEEDS_ARTIFACT_TYPE1 + "\n"//
+                + "+ " + NEEDS_ARTIFACT_TYPE2 + "\n";
+    }
+
+    private void assertAllImporterEventsForLegacyItemCalled()
+    {
+        final InOrder inOrder = inOrder(this.listenerMock);
```

Comment:
InOrder might be a bit too paranoid and could cause false negatives.

Reply:
But isn't this basically a protocol?
It is certainly not stateless with "begin" and "end".